test/unit/ardb_tests.rb

```diff
@@ -1,67 +1,125 @@
 require 'assert'
 require 'ardb'
 
+require 'logger'
+
 module Ardb
 
   class UnitTests < Assert::Context
     desc "Ardb"
     setup do
-      @orig_ar_logger = ActiveRecord::Base.logger
-      Adapter.reset
-
       @module = Ardb
     end
-    teardown do
-      Adapter.reset
-      ActiveRecord::Base.logger = @orig_ar_logger
-    end
     subject{ @module }
 
     should have_imeths :config, :configure, :adapter, :validate!, :init
     should have_imeths :escape_like_pattern
 
+    should "default the db file env var" do
+      skip # this won't pass while the `test/helper` overwrites it
+      assert_equal 'config/db', ENV['ARDB_DB_FILE']
+    end
+
     should "return its `Config` class with the `config` method" do
       assert_same Config, subject.config
     end
 
-    should "complain if init'ing and not all configs are set" do
-      orig_adapter = Ardb.config.db.adapter
-      Ardb.config.db.adapter = nil
-      assert_raises(NotConfiguredError){ subject.init }
-      Ardb.config.db.adapter = orig_adapter
+  end
+
+  class InitMethodSetupTests < UnitTests
+    setup do
+      @orig_env_pwd             = ENV['PWD']
+      @orig_env_ardb_db_file    = ENV['ARDB_DB_FILE']
+      @orig_ar_logger           = ActiveRecord::Base.logger
+      @orig_ardb_config_options = Config.to_hash
+      Config.reset
+      Adapter.reset
+
+      ENV['ARDB_DB_FILE'] = 'test/support/require_test_db_file'
+      Ardb.configure do |c|
+        c.root_path    = TMP_PATH
+        c.logger       = Logger.new(STDOUT)
+        c.db.adapter   = 'postgresql'
+        c.db.database  = Factory.string
+      end
+
+      @ar_establish_connection_called_with = nil
+      Assert.stub(ActiveRecord::Base, :establish_connection) do |options|
+        @ar_establish_connection_called_with = options
+      end
+    end
+    teardown do
+      Adapter.reset
+      Config.apply(@orig_ardb_config_options)
+      ActiveRecord::Base.logger = @orig_ar_logger
+      ENV['ARDB_DB_FILE']       = @orig_env_ardb_db_file
+      ENV['PWD']                = @orig_env_pwd
+    end
+
+  end
+
+  class InitMethodTests < InitMethodSetupTests
+    desc "the `init` method"
```

Comment:
@jcredding just say "`init` method" here.  Otherwise the test desc would read "Ardb the `init` method...".